hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/WatermarkKeysAssignerTest.java

```diff
@@ -240,14 +240,92 @@ public void when_streamToStreamJoinIsPresent_then_keyWasPropagated() {
 
         assertThat(finalOptRel).isInstanceOf(StreamToStreamJoinPhysicalRel.class);
 
-        // Watermark key was propagated to UnionPhysicalRel
+        // Watermark key was propagated to StreamToStreamJoinPhysicalRel.
         Map<Integer, MutableByte> map = keysAssigner.getWatermarkedFieldsKey(finalOptRel);
         assertThat(map).isNotNull();
         assertThat(map).isNotEmpty();
         assertThat(map.get(0)).isNotNull();
         assertThat(map.get(0).getValue()).isEqualTo((byte) 0);
     }
 
+    @Test
+    public void when_s2sJoinWithWindowAggregationIsPresent_then_keyWasPropagated() {
+        NodeEngine nodeEngine = getNodeEngine(instance());
+        TableResolverImpl resolver = new TableResolverImpl(
+                nodeEngine,
+                new RelationsStorage(nodeEngine),
+                new SqlConnectorCache(nodeEngine));
+
+        String stream = "s";
+        TestStreamSqlConnector.create(
+                instance().getSql(),
+                stream,
+                singletonList("a"),
+                singletonList(BIGINT),
+                row(1L)
+        );
+
+        assertInstanceOf(TestAbstractSqlConnector.TestTable.class, resolver.getTables().get(0));
```

Comment:
this looks like a sanity check for the test setup ("given"). Such checks usually don't add value and complicate the test. Do you expect other objects present in `TableResolverImpl`?

Reply:
BTW, AssertJ has `assertThat(...).isInstanceOf(...)`, better to keep one style of asserts in single test